src/openforms/forms/models/form_variable.py

```diff
@@ -222,6 +222,9 @@ class Meta:
     def __str__(self):
         return _("Form variable '{key}'").format(key=self.key)
 
+    def matches_json_schema(self, json_schema: dict[str, Any]) -> bool:
+        pass
+
```

Comment:
I initially had something like `as_json_schema(self) -> dict[str, Any]` in mind. But as discussed to have the compatibility logic simple at first, producing JSON Schemas for any variable isn't possible in all cases. So this method will take care of implementing custom logic instead of comparing `as_json_schema() == sub_json_schema`